### PR TITLE
cardano-cli: Add minted assets to transaction-view

### PR DESCRIPTION
**Pull request by cblp**
Task: #2533

### CODE REVIEW

File: cardano-cli/cardano-cli.cabal [hunk 1]
```diff
@@ -15,7 +15,8 @@ Flag unexpected_thunks
   Description:   Turn on unexpected thunks checks
   Default:       False
 
-common base                         { build-depends: base                             >= 4.14       && < 4.15     }
```
Comment (Jimbo4350) [inline cardano-cli/cardano-cli.cabal:18]: :+1:

**Review by Jimbo4350 — COMMENTED** [review]
I've compared the test failure outputs:
This PR the test failure output is:
```
Process exited with non-zero exit-code
    ━━━━ stdout ━━━━
    --- -       2021-04-07 11:51:46.236794269 +0100
    +++ test/data/golden/shelley/transaction-view.out   2021-04-07 11:43:56.082976405 +0100
    @@ -3,7 +3,7 @@
     era: Shelley
     fee: 32
     inputs:
    -- fe4dd07fb576bff960d6e066eade5b26cdb5afebe29f76ea58d0a098bce5d891#29
    +- fe5dd07fb576bff960d6e066eade5b26cdb5afebe29f76ea58d0a098bce5d891#29
     metadata hash: null
     outputs:
     - address:
    
    ━━━━ stderr ━━━━
```
On master the test output failure is:
```
       ┃       │ ━━━ Failed (- lhs) (+ rhs) ━━━
       ┃       │   [
       ┃       │     "auxiliary data: null"
       ┃       │   , "certificates: []"
       ┃       │   , "era: Shelley"
       ┃       │   , "fee: 32"
       ┃       │   , "inputs:"
       ┃       │ - , "- fe4dd07fb576bff960d6e066eade5b26cdb5afebe29f76ea58d0a098bce5d891#29"
       ┃       │ + , "- fe5dd07fb576bff960d6e066eade5b26cdb5afebe29f76ea58d0a098bce5d891#29"
       ┃       │   , "metadata hash: null"
       ┃       │   , "outputs:"
       ┃       │   , "- address:"
       ┃       │   , "    Bech32: addr1v9wmu83pzajplrtpsq6tsqdgwr98x888trpmah2u0ezznsge7del3"
       ┃       │   , "    credential:"
       ┃       │   , "      key hash: 5dbe1e2117641f8d618034b801a870ca731ce758c3bedd5c7e4429c1"
       ┃       │   , "    network: Mainnet"
       ┃       │   , "    stake reference: StakeRefNull"
       ┃       │   , "  amount: 31"
       ┃       │   , "time to live: 33"
       ┃       │   , "update: null"
       ┃       │   , "withdrawals: []"
       ┃       │   , ""
       ┃       │   ]
```
I can't see a clear benefit or advantage of using `diff`. This introduces an additional dependency on the user and gives essentially the same test failure output.

**Comment by cblp** [comment]
@Jimbo4350 
When list size differ, hedgehog shows them in full.

**Comment by Jimbo4350** [comment]
> @Jimbo4350
> When list size differ, hedgehog shows them in full.

In what instance for a golden test will the list size differ? Can you give me a concrete example?

File: cardano-cli/test/Test/Golden/Diff.hs [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+import           Cardano.Prelude
+import           Prelude (String)
+import qualified Prelude
+
+import           Data.Algorithm.Diff (PolyDiff (Both), getGroupedDiff)
+import           Data.Algorithm.DiffOutput (ppDiff)
+import           Hedgehog (MonadTest)
+import           Hedgehog.Extras.Test.Base (failMessage)
+
+diffVsFile
```
Comment (Jimbo4350) [inline cardano-cli/test/Test/Golden/Diff.hs:12]: Let's move this to `cardano-cli/test/Test/Utilities.hs`

File: cardano-cli/test/Test/Golden/Diff.hs [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+  :: (MonadIO m, MonadTest m)
+  => String   -- ^ actual content
+  -> FilePath -- ^ reference file
+  -> m ()
+diffVsFile actualContent referenceFile =
```
Comment (Jimbo4350) [inline cardano-cli/test/Test/Golden/Diff.hs:17]: :+1:

**Review by Jimbo4350 — CHANGES_REQUESTED** [review]
Nice! One module needs to be moved and I will approve after that.

File: cardano-cli/test/Test/Golden/Diff.hs (hunk 2 above)
Comment (cblp) [reply cardano-cli/test/Test/Golden/Diff.hs]: done

**Review by Jimbo4350 — APPROVED** [review]
LGTM!

**Comment by cblp** [comment]
bors merge

**Pull request merged**